base_layer/core/src/transactions/transaction.rs

```diff
@@ -638,6 +649,31 @@ impl Transaction {
     pub fn min_spendable_height(&self) -> u64 {
         max(self.max_kernel_timelock(), self.max_input_maturity())
     }
+
+    // Note this will also do cut-through
+    pub fn add_no_cut_through(mut self, other: Self) -> Self {
+        self.offset = self.offset + other.offset;
+        let (mut inputs, mut outputs, mut kernels) = other.body.dissolve();
+        self.body.add_inputs(&mut inputs);
+        self.body.add_outputs(&mut outputs);
+        self.body.add_kernels(&mut kernels);
+        self
+    }
+}
+
+impl Add for Transaction {
+    type Output = Self;
+
+    // Note this will also do cut-through
+    fn add(mut self, other: Self) -> Self {
+        self.offset = self.offset + other.offset;
+        let (mut inputs, mut outputs, mut kernels) = other.body.dissolve();
+        self.body.add_inputs(&mut inputs);
+        self.body.add_outputs(&mut outputs);
+        self.body.add_kernels(&mut kernels);
```

Comment:
Repeated code. Can you call the function above?

Reply:
done